static/sharing.js

```diff
@@ -30,6 +30,25 @@ var shortenURL = require('./urlshorten-' + options.urlShortenService);
 var Components = require('components');
 var url = require('./url');
 
+var shareServices = {
+    twitter: {
+        cssClass: 'share-twitter',
+        getLink: function (title, url) {
+            return "https://twitter.com/intent/tweet?text=" +
+                encodeURIComponent(title) + '&url=' + encodeURIComponent(url) + '&via=mattgodbolt';
+        },
+        text: 'Tweet'
+    },
+    reddit: {
```

Comment:
I just played about with the sharing. It makes sense for "Full" and "Short" URLs. But if you have "embed" or "embed readonly" selected you end up trying to post an HTML fragment to either reddit or twitter (e.g.

https://www.reddit.com/submit?url=%3Ciframe%20width%3D%22800px%22%20height%3D%22200px%22%20src%3D%22http%3A%2F%2Flocalhost%3A10240%2Fembed-ro%23g%3A!((g%3A!((g%3A!((h%3AcodeEditor%2Ci%3A(j%3A1%2Clang%3Ac%252B%252B%2Csource%3A%27%2523include%2B%253Ctype_traits%253E%250A%2523include%2B%253Cutility%253E%250A%250Atemplate%253Csize_t%2BI%253E%250Astruct%2BIdx%257B%257D%253B%250A%250Atemplate%253Csize_t%2BCount%253E%250Astruct%2Bmake_idx%250A%257B%250A%2B%2B%2B%2Btemplate%253Ctypename...%2BTs%253E%250A%2B%2B%2B%2Bstruct%2Bidx_t%257B%250A%2B%2B%2B%2B%2B%2B%2B%2Bstatic%2Bconstexpr%2Bvoid%2Btest()%2B%257B%257D%250A%2B%2B%2B%2B%257D%253B%250A%250A%2B%2B%2B%2Btemplate%253Csize_t...%2BIs%253E%250A%2B%2B%2B%2Bstatic%2Bconstexpr%2Bauto%2Bget_type_impl(std%3A%3Aindex_sequence%253CIs...%253E)%250A%2B%2B%2B%2B%257B%250A%2B%2B%2B%2B%2B%2B%2B%2Breturn%2Bidx_t%253CIdx%253CIs%253E...%253E%257B%257D%253B%250A%2B%2B%2B%2B%257D%250A%2B%2B%2B%2Busing%2Btype%2B%253D%2Bdecltype(get_type_impl(std%3A%3Amake_index_sequence%253CCount%253E%257B%257D))%253B%250A%257D%253B%250A%250Aint%2Bmain()%250A%257B%250A%2B%2B%2B%2Busing%2Blarge_template%2B%253D%2Bmake_idx%253C10000%253E%3A%3Atype%253B%250A%2B%2B%2B%2Blarge_template%3A%3Atest()%253B%250A%257D%27)%2Cl%3A%275%27%2Cn%3A%270%27%2Co%3A%27C%252B%252B%2Bsource%2B%25231%27%2Ct%3A%270%27))%2Ck%3A50%2Cl%3A%274%27%2Cn%3A%270%27%2Co%3A%27%27%2Cs%3A0%2Ct%3A%270%27)%2C(g%3A!((g%3A!((h%3Acompiler%2Ci%3A(compiler%3Ag81%2Cfilters%3A(b%3A%270%27%2Cbinary%3A%271%27%2CcommentOnly%3A%270%27%2Cdemangle%3A%270%27%2Cdirectives%3A%270%27%2Cexecute%3A%271%27%2Cintel%3A%270%27%2Ctrim%3A%270%27)%2Clang%3Ac%252B%252B%2Clibs%3A!((name%3Acmcstl2%2Cver%3Atrunk))%2Coptions%3A%27-std%253Dc%252B%252B1z%27%2Csource%3A1)%2Cl%3A%275%27%2Cn%3A%270%27%2Co%3A%27x86-64%2Bgcc%2B8.1%2B(Editor%2B%25231%2C%2BCompiler%2B%25231)%2BC%252B%252B%27%2Ct%3A%270%27))%2Ck%3A50%2Cl%3A%274%27%2Cm%3A79.83706720977597%2Cn%3A%270%27%2Co%3A%27%27%2Cs%3A0%2Ct%3A%270%27)%2C(g%3A!((h%3Aoutput%2Ci%3A(compiler%3A1%2Ceditor%3A1%2Cwrap%3A%271%27)%2Cl%3A%275%27%2Cn%3A%270%27%2Co%3A%27%25231%2Bwith%2Bx86-64%2Bgcc%2B8.1%27%2Ct%3A%270%27))%2Cheader%3A()%2Cl%3A%274%27%2Cm%3A20.16293279022403%2Cn%3A%270%27%2Co%3A%27%27%2Cs%3A0%2Ct%3A%270%27))%2Ck%3A50%2Cl%3A%273%27%2Cn%3A%270%27%2Co%3A%27%27%2Ct%3A%270%27))%2Cl%3A%272%27%2Cn%3A%270%27%2Co%3A%27%27%2Ct%3A%270%27))%2Cversion%3A4%22%3E%3C%2Fiframe%3E&title=Compiler%20Explorer or similar)

...and that's not great. It's probably not a real problem, but worth noting. It would require more sophistication to fix, and I don't know what the solution should be.

Reply:
We could probably disable those buttons when any form of embed binds are selected. Something like graying them out and adding a title explaining why they're disabled would work pretty well in my opinion

Reply:
Sounds like a plan if it's easy to do!